backend/app/routes.py

```diff
@@ -41,8 +46,43 @@ def publish_message_async(data):
     """Function to publish messages asynchronously"""
     executor.submit(publish_message, data)
 
+limiter = Limiter(get_remote_address, app=app, default_limits=["200 per day", "50 per hour"])
+
+@app.route('/register', methods=['POST'])
+@limiter.limit(limit_value="5 per minute")  # Limit registration attempts
+def register():
+    """Register a new user"""
+    username = request.json.get('username')
+    password = request.json.get('password')
+
+    if User.query.filter_by(username=username).first():
+        return jsonify({"msg": "User already exists"}), 400
+
+    # Create new user and add to the database
+    new_user = User(username=username, password=password)
+    db.session.add(new_user)
+    db.session.commit()
+    return jsonify({"msg": "User registered successfully"}), 201
+
+# User login
+@app.route('/login', methods=['POST'])
+@limiter.limit(limit_value="10 per minute")  # Limit login attempts
+def login():
+    """Login an authenticated user"""
+    username = request.json.get('username')
+    password = request.json.get('password')
+
+    user = User.query.filter_by(username=username).first()
+    if user and user.check_password(password):
+        # Create access token
+        access_token = create_access_token(identity=username)
+        return jsonify({"access_token": access_token, "msg": "Logged In Successfully"}), 200
+    else:
+        return jsonify({"msg": "Invalid credentials"}), 401
+
 # Routes
 @app.route("/create", methods=["POST"])
+@jwt_required()
 def create():
```

Comment:
Looking at your endpoint implementations would be nice to refactor it using Marshmallow Schema, Flask_restfull and using existing SQLAlchemy.

- [ ] Standardize your endpoints using flask_restfull https://flask-restful.readthedocs.io/en/latest/ and Marshmallow schema https://marshmallow.readthedocs.io/en/stable/

Reply:
I have completed all the above suggestions

Reply:
@abeni-al7 great job so far, impressive learning curve.

Reply:
@abeni-al7 was thinking to add more attributes to a user model as this things is going to evolve to a full-fledged user module. It would be nice if we add first_name, middle_name, last_name, type(make this string enum like admin, editor,), status (boolean true for active, false for inactive), email and phone_number.

- [ ] Add more attributes to a user model.

Reply:
> @abeni-al7 great job so far, impressive learning curve.

Thank you very much

Reply:
@abeni-al7 now prove that what you did is correct by writing an integration test for all your enpoints. Later we will include this test to a pipeline once done with github workflow. 

- [ ] Write integration test for all endpoints better to define all your tests under a folder called tests/integration/user_test.py...later as our system evolves we will also include unit tests.

Reply:
Also try to follow our commit format, I noticed as you aren't following commit message formats in your recent commits. 

- [ ] Follow commit formats